lib/core/util.js

```diff
@@ -26,6 +26,59 @@ function isBlobLike (object) {
   )
 }
 
+function isDate (val) {
+  return toString.call(val) === '[object Date]'
+}
+
+function isObject (val) {
+  return val !== null && typeof val === 'object'
+}
+
+// this escapes all non-uri friendly characters, but restores : which is commonly used for date formatting
+function encode (val) {
+  return encodeURIComponent(val).replace(/%3A/gi, ':')
```

Comment:
Why restore `:`? I don't understand the use case?

Reply:
Escaping `:` is going to mess up date format and complicate rather frequent use-case.
I'm having second thoughts about escaping anything, TBH, especially since Undici is not supposed to be used from browser. What do you think about removing escaping completely and let user do it if they need to?

Reply:
I'm ok with removing escaping. What does got do?

Reply:
I think we actually need to escape or we might end up with invalid URLs, e.g. if they contain `?`, `&` etc...

Reply:
I think `encodeURIComponent` is sufficient for now.

Reply:
Just need to clarify in the docs that we expect unencoded keys and values.

Reply:
I'll compare with got implementation later today.

Reply:
What I'm concerned about is that there is not much of an escape hatch there. If receiving API is not capable of parsing escaped query params, user will be forced to just hardcode entire URL, which we are not encoding; and it is likely going to be confusing for the user, why are we treating query params differently.

Reply:
This is why I'd prefer not to have these kinds of things in undici. There is no perfect solution. I think not escaping is worse than escaping. If the receiving API needs something else then the user can always build the path themselves.

Reply:
got is using URLSearchParams, and there are already issues with that: https://github.com/sindresorhus/got/issues/1180
If we go with escaping, I would suggest following axios example instead of got - they were around for much longer, and had all the time in the world to work out the kinks of real world usage.

Reply:
I'm not sure I understand the use case. Is it that some server implementations don't decode query parameters at all?

Reply:
I'd prefer if we just keep it simple and use `encodeURIComponent`.

Reply:
I agree

Reply:
done